fluent-bundle/src/builtins.ts

```diff
@@ -37,12 +54,35 @@ export function NUMBER(
   }
 
   if (arg instanceof FluentNumber || arg instanceof FluentDateTime) {
-    return new FluentNumber(arg.valueOf(), { ...arg.opts, ...values(opts) });
+    return new FluentNumber(arg.valueOf(), {
+      ...arg.opts,
+      ...values(opts, NUMBER_ALLOWED)
+    });
   }
 
   throw new TypeError("Invalid argument to NUMBER");
 }
 
+const DATETIME_ALLOWED = [
+  "dateStyle",
+  "timeStyle",
+  "fractionalSecondDigits",
+  "calendar",
+  "dayPeriod",
+  "numberingSystem",
```

Comment:
Yeah, I'm torn on this one, too. I'd go with restrictive for now, and not allow this one as well (same for `NUMBER`).